File: app/src/main/java/com/xuie/android/ui/coordinatorLayout/CoordinatorLayoutActivity.java
package com.xuie.android.ui.coordinatorLayout;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import com.xuie.android.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public class CoordinatorLayoutActivity extends AppCompatActivity {

    @BindView(R.id.toolbar) Toolbar toolbar;
    @BindView(R.id.fab) FloatingActionButton fab;

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_coordinator_layout);
        ButterKnife.bind(this);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        fab.setOnClickListener(view -> Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG).setAction("Action", null).show());
    }

    @Override public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/xuk/android/ui/coordinator/CoordinatorLayoutActivity.kt
package xuk.android.ui.coordinator

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.snackbar.Snackbar
import xuk.android.databinding.ActivityCoordinatorLayoutBinding

/**
 * @author Jie Xu
 */
class CoordinatorLayoutActivity : AppCompatActivity() {

  private lateinit var binding: ActivityCoordinatorLayoutBinding

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    ActivityCoordinatorLayoutBinding.inflate(layoutInflater).apply {
      binding = this
      setContentView(root)

      setSupportActionBar(toolbar)
      supportActionBar?.setDisplayHomeAsUpEnabled(true)

      fab.setOnClickListener { view ->
        Snackbar.make(
          view,
          "Replace with your own action",
          Snackbar.LENGTH_LONG
        ).setAction("Action", null).show()
      }
    }
  }

  override fun onOptionsItemSelected(item: MenuItem): Boolean {
    if (item.itemId == android.R.id.home) {
      onBackPressed()
    }
    return super.onOptionsItemSelected(item)
  }
}
